### ISSUE TITLE
redux persist - firebase not persisting to realtime database

### ISSUE THREAD

**Issue by kstott** [prescottprue/react-redux-firebase#526]
I've implemented createStore.js per the example. And used the store created in an existing redux-persist react-native app. It persists the data locally, but it does not transfer to the realtime database. I am able to login successfully. I am using react-native-firebase. 

The realtime database rule is:

`{
  "rules": {
    ".read": true,
    ".write": true
  }
}`

createStore.js is:

`
```
....
import { createStore, combineReducers, compose, applyMiddleware } from 'redux'
import { reactReduxFirebase, firebaseReducer } from 'react-redux-firebase'
import firebase from 'react-native-firebase'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web and AsyncStorage for react-native
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import logger from 'redux-logger'

const persistConfig = {
    key: 'root',
    storage,
    blacklist: ['session'],
    stateReconciler: autoMergeLevel2,
    debug: true
};

const rrfConfig = {
    userProfile: 'users',
    enableRedirectHandling: false,
    enableLogging: true
    //useFirestoreForProfile: true // Firestore for Profile instead of Realtime DB
};

const firebaseConfig =  {
...
};

export default (initialState = {}, history) => {
```

    // Initialize firebase instance
    firebase.initializeApp(firebaseConfig)

    // Add reactReduxFirebase store enhancer when making store creator
    const createStoreWithFirebase = compose(
        reactReduxFirebase(firebase, rrfConfig), // firebase instance as first argument
    )(createStore);

    // Add firebase to reducers (uses persistReducer and hardSet)
    const rootReducer = combineReducers({
        firebase: persistReducer(
            { key: 'firepersist', storage: storage},
            firebaseReducer
        ),
        authentication,
        usage,
        boards,
        media,
        session,
        cell,
        ...createForms({
            display,
            authoring,
            textToSpeech,
            touch,
            switchScanning,
            phraseMode,
            addresses,
            wordPrediction,
            other
        })
    });

    const persistedReducer = persistReducer(persistConfig, rootReducer);
    const store = createStore(
        persistedReducer,
        initialState,
        applyMiddleware(logger));
    const persistor = persistStore(store);

    return { store, persistor }
}
```
`

package.json:

`"dependencies": {
    "react-native": "^0.56.0",
    "react-native-firebase": "^4.3.8",
    "react-redux": "^5.0.7",
    "react-redux-firebase": "^2.1.8",
    "redux": "^4.0.0",
    "redux-persist": "^5.6.12",
},`

I haven't touched any other code (other than adding the rnfirebase pod to Xcode). I made the assumption that I only had to modify the store creation. 
 
Any assistance would be appreciated.
 
Ken

**Comment by kstott** [comment]
If I am just misunderstanding the configuration - just point me to the relevant docs and I will figure it out. I considered that firebaseconnect had something to do with it. Perhaps I needed to use that in mapstatetoprops. But that looked more like a pattern that replaced redux persist.

**Comment by prescottprue** [comment]
@kstott Maybe I am understanding the question, but it seems like you may be misinterpreting how data is persisted. `react-redux-firebase` syncs data into your redux store *from* Firebase, there is not a mechanism that automatically syncs your store changes *to* Firebase.

You make listeners with `firebaseConnect` or `watchEvent` manually, and those write data to redux as explained [in the queries section of the docs](http://docs.react-redux-firebase.com/history/v2.0.0/docs/queries.html). Data can be written to Firebase as done with the JS sdk by default or by using exposed methods which dispatch actions, which is [described in the actions section of the docs](http://docs.react-redux-firebase.com/history/v2.0.0/docs/recipes/actions.html).

There is a diagram that may help visualize [in 4. of the FAQ section](http://docs.react-redux-firebase.com/history/v2.0.0/docs/FAQ.html#faq).

**Comment by kstott** [comment]
I figured I misunderstood. Thanks for the links. Let me figure it out.

Sent from my iPhone

On Aug 19, 2018, at 7:37 PM, Scott <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #526<https://github.com/prescottprue/react-redux-firebase/issues/526>.

**Comment by prescottprue** [comment]
@kstott Sounds good. Feel free to reach out over [gitter](gitter.im/redux-firebase/Lobby) if you have any questions 😄.
